protocol/messenger_communities.go

```diff
@@ -2437,129 +2414,178 @@ func (m *Messenger) RequestImportDiscordCommunity(request *requests.ImportDiscor
+					// import has finished)
+					communityWithCategories, changes, err := m.communitiesManager.CreateCategory(createCommunityCategoryRequest, false)
+					if err != nil {
+						m.cleanUpImport(communityID)
+						importProgress.AddTaskError(discord.CommunityCreationTask, discord.Error(err.Error()))
+						importProgress.StopTask(discord.CommunityCreationTask)
+						progressUpdates <- importProgress
+						return
+					}
+					discordCommunity = communityWithCategories
+					// This looks like we keep overriding the same field but there's
+					// only one `CategoriesAdded` change at this point.
+					for _, addedCategory := range changes.CategoriesAdded {
+						processedCategoriesIds[category.ID] = addedCategory.CategoryId
+					}
+				}
 			}
 
-			// We call `CreateChat` on `communitiesManager` directly to get more control
-			// over whether we want to publish the updated community description.
-			communityWithChats, changes, err := m.communitiesManager.CreateChat(discordCommunity.ID(), communityChat, false)
-			if err != nil {
-				m.cleanUpImport(communityID)
-				errmsg := err.Error()
-				if _errors.Is(err, communities.ErrInvalidCommunityDescriptionDuplicatedName) {
-					errmsg = fmt.Sprintf("Couldn't create channel '%s': %s", communityChat.Identity.DisplayName, err.Error())
-				}
-				importProgress.AddTaskError(discord.ChannelsCreationTask, discord.Error(errmsg))
-				importProgress.StopTask(discord.ChannelsCreationTask)
+			progressValue := calculateProgress(i+1, totalImportChunkCount, (1 / 2))
+			importProgress.UpdateTaskProgress(discord.ChannelsCreationTask, progressValue)
+
+			progressUpdates <- importProgress
+
+			if m.DiscordImportMarkedAsCancelled(communityID) {
+				importProgress.StopTask(discord.CommunityCreationTask)
 				progressUpdates <- importProgress
+				cancel <- communityID
 				return
 			}
-			discordCommunity = communityWithChats
-
-			// This looks like we keep overriding the chat id value
-			// as we iterate over `ChatsAdded`, however at this point we
-			// know there was only a single such change (and it's a map)
-			for chatID, chat := range changes.ChatsAdded {
-				c := CreateCommunityChat(communityID, chatID, chat, m.getTimesource())
-				chatsToSave = append(chatsToSave, c)
-				processedChannelIds[channel.Channel.ID] = c.ID
+
+			messagesToSave := make(map[string]*common.Message, 0)
+			pinMessagesToSave := make([]*common.PinMessage, 0)
+			authorProfilesToSave := make(map[string]*protobuf.DiscordMessageAuthor, 0)
+			messageAttachmentsToDownload := make([]*protobuf.DiscordMessageAttachment, 0)
+
+			// Save to access the first item here as we process
+			// exported data by files which only holds a single channel
+			channel := exportData.ExportedData[0]
+			chatIDs := discordCommunity.ChatIDs()
+
+			exists := false
+			for _, chatID := range chatIDs {
+				if strings.HasSuffix(chatID, channel.Channel.ID) {
+					exists = true
+					break
+				}
+			}
```

Comment:
@cammellos here's how we check for any chat that has already been created during the import